### PR TITLE
Create StaticHttpContentFactory and other cleanups

### PR DESCRIPTION
**Pull request by lachlan-roberts**
- Create `StaticHttpContentFactory` for stylesheet.
- Move all HttpContent related classes into separate package.
- Fix potential bug with `FileMappingHttpContentFactory`.
- Other minor cleanups.

### CODE REVIEW

File: jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/StaticHttpContentFactory.java
Comment (gregw) [inline jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/StaticHttpContentFactory.java]: This class needs javadoc explaining what it is....  it also probably needs a better name as it's not for static resources.    It is a virtual directory entry resource, as it fakes an entry in every directory.  So:
 + VirtualHttpContentFactory?
 + PhantomHttpContentFactory?
 + PerDirectoryHttpContentFactory?

File: jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/StaticHttpContentFactory.java
Comment (gregw) [inline jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/StaticHttpContentFactory.java]: This is not necessarily a style sheet.  This technique might also be used for js or icons

File: jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/StaticHttpContentFactory.java
Comment (gregw) [inline jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/StaticHttpContentFactory.java]: You should allow the contentType to be passed into the constructor

File: jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/StaticHttpContentFactory.java
Comment (gregw) [inline jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/content/StaticHttpContentFactory.java]: EOL

**Review by gregw — CHANGES_REQUESTED** [review]

**Comment by lachlan-roberts** [comment]
Not sure whats going on here, all these failing tests pass for me locally and I can step though and everything looks normal.

**Comment by joakime** [comment]
@lachlan-roberts something funny is going on with the Resource.getFileName() on CI.

On a local run, I see ...

```
2022-12-21 14:19:20.428:DEBUG:oejes.ServletHolder:qtp829149076-36: Servlet.init null for org.eclipse.jetty.ee10.servlet.DefaultServlet-20b2475a
2022-12-21 14:19:20.437:DEBUG:oejes.DefaultServlet:qtp829149076-36:   .baseResource = file:///home/joakim/code/jetty/jetty.project-12.0.x/jetty-ee10/jetty-ee10-servlet/target/tests/oejes.DefaultServletTest/testListingProperUrlEncoding/docroot/
2022-12-21 14:19:20.437:DEBUG:oejes.DefaultServlet:qtp829149076-36:   .resourceFactory = null
2022-12-21 14:19:20.437:DEBUG:oejes.DefaultServlet:qtp829149076-36:   .resourceService = org.eclipse.jetty.ee10.servlet.DefaultServlet$ServletResourceService@13154c37(contentFactory=PreCompressedHttpContentFactory@362df692[org.eclipse.jetty.http.content.VirtualHttpContentFactory@4a879256(factory=ResourceContentFactory[org.eclipse.jetty.util.resource.ResourceFactory$1@3721d992]@241977500, resource=file:/home/joakim/code/jetty/jetty.project-12.0.x/jetty-core/jetty-server/target/classes/org/eclipse/jetty/server/jetty-dir.css, matchSuffix=/jetty-dir.css, contentType=text/css),[]], dirAllowed=true, redirectWelcome=false)
2022-12-21 14:19:20.438:DEBUG:oejes.DefaultServlet:qtp829149076-36:   .isPathInfoOnly = false
2022-12-21 14:19:20.438:DEBUG:oejes.DefaultServlet:qtp829149076-36:   .welcomeExactServlets = false
2022-12-21 14:19:20.438:DEBUG:oejes.DefaultServlet:qtp829149076-36:   .welcomeServlets = false
2022-12-21 14:19:22.477:DEBUG:oejes.DefaultServlet:qtp829149076-36: doGet(req=org.eclipse.jetty.ee10.servlet.ServletContextRequest$ServletApiRequest@4d12235d, resp=org.eclipse.jetty.ee10.servlet.ServletContextResponse$ServletApiResponse@2a907f5d) pathInContext=/dir/, included=false
2022-12-21 14:20:35.070:DEBUG:oejes.DefaultServlet:qtp829149076-36: content = null
```

But on CI (with DEBUG enabled) I see ...

```
2022-12-21 18:47:17.767:DEBUG:oejes.ServletHolder:qtp1871048194-1492: Servlet.init null for org.eclipse.jetty.ee10.servlet.DefaultServlet-269222ae
2022-12-21 18:47:17.768:DEBUG:oejes.DefaultServlet:qtp1871048194-1492:   .baseResource = file:///home/jenkins/agent/workspace/jetty.project_PR-9043/jetty-ee10/jetty-ee10-servlet/target/tests/oejes.DefaultServletTest/testListingProperUrlEncoding/docroot/
2022-12-21 18:47:17.768:DEBUG:oejes.DefaultServlet:qtp1871048194-1492:   .resourceFactory = null
2022-12-21 18:47:17.768:DEBUG:oejes.DefaultServlet:qtp1871048194-1492:   .resourceService = org.eclipse.jetty.ee10.servlet.DefaultServlet$ServletResourceService@12f62694(contentFactory=PreCompressedHttpContentFactory@2a2b201a[org.eclipse.jetty.http.content.VirtualHttpContentFactory@6d5678dc(factory=ResourceContentFactory[org.eclipse.jetty.util.resource.ResourceFactory$1@3a456b4a]@421805180, resource=jar:file:///home/jenkins/agent/workspace/jetty.project_PR-9043/jetty-core/jetty-server/target/jetty-server-12.0.0-SNAPSHOT.jar!/org/eclipse/jetty/server/jetty-dir.css, matchSuffix=/, contentType=text/css),[]], dirAllowed=true, redirectWelcome=false)
2022-12-21 18:47:17.768:DEBUG:oejes.DefaultServlet:qtp1871048194-1492:   .isPathInfoOnly = false
2022-12-21 18:47:17.768:DEBUG:oejes.DefaultServlet:qtp1871048194-1492:   .welcomeExactServlets = false
2022-12-21 18:47:17.768:DEBUG:oejes.DefaultServlet:qtp1871048194-1492:   .welcomeServlets = false
2022-12-21 18:47:17.768:DEBUG:oejes.DefaultServlet:qtp1871048194-1492: doGet(req=org.eclipse.jetty.ee10.servlet.ServletContextRequest$ServletApiRequest@499db7e8, resp=org.eclipse.jetty.ee10.servlet.ServletContextResponse$ServletApiResponse@17d3a69f) pathInContext=/dir/, included=false
2022-12-21 18:47:17.768:DEBUG:oejes.DefaultServlet:qtp1871048194-1492: content = CompressedFormatsHttpContent@2facbfc5[ResourceHttpContent@3ea28e41{r=jar:file:///home/jenkins/agent/workspace/jetty.project_PR-9043/jetty-core/jetty-server/target/jetty-server-12.0.0-SNAPSHOT.jar!/org/eclipse/jetty/server/jetty-dir.css,ct=text/css}]
```

The CI run for `pathInContext = "/dir/"` results in an `HttpContent` of `CompressedFormatsHttpContent@2facbfc5[ResourceHttpContent@3ea28e41{r=jar:file:///home/jenkins/agent/workspace/jetty.project_PR-9043/jetty-core/jetty-server/target/jetty-server-12.0.0-SNAPSHOT.jar!/org/eclipse/jetty/server/jetty-dir.css,ct=text/css}]`

But on the local run, the same `pathInContext = "/dir/"` results in a `null` `HttpContent`.

If we look at the state of the `VirtualHttpContentFactory` we can see the following

Env | VirtualHttpContentFactory Value
---- | ----
CI | resource=jar:file:///home/jenkins/agent/workspace/jetty.project_PR-9043/jetty-core/jetty-server/target/jetty-server-12.0.0-SNAPSHOT.jar!/org/eclipse/jetty/server/jetty-dir.css
Local | resource=file:/home/joakim/code/jetty/jetty.project-12.0.x/jetty-core/jetty-server/target/classes/org/eclipse/jetty/server/jetty-dir.css
CI | matchSuffix=/
Local | matchSuffix=/jetty-dir.css
CI | contentType=text/css
Local | contentType=text/css

The type of resource is different, `file://` vs `jar:file://` and the "matchSuffix" is different.

**Comment by joakime** [comment]
@lachlan-roberts I think I got a solution.

The `MemoryResource` doesn't have adequate testing, nor did the `FileID.getFileName(URI)` that it relies on.
The fix in commit 0485fdf2f36673e4fbf697ccdf2ee1e6d82506ce should take care of this issue.

File: jetty-ee10/jetty-ee10-servlet/src/test/resources/jetty-logging.properties
Comment (joakime) [inline jetty-ee10/jetty-ee10-servlet/src/test/resources/jetty-logging.properties]: This will need to be reverted once we have our first green build

File: jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DefaultServletTest.java
Comment (joakime) [inline jetty-ee10/jetty-ee10-servlet/src/test/java/org/eclipse/jetty/ee10/servlet/DefaultServletTest.java]: This test directory tree dump should be removed once we have our first green build

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java
Comment (joakime) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java:261]: This is a legit fix to the Deprecation warning.

**Comment by joakime** [comment]
@lachlan-roberts this PR is now green

**Comment by lachlan-roberts** [comment]
thanks @joakime, I will merge once I get an approved review

**Review by joakime — APPROVED** [review]

**Pull request merged**
